cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_nodegroup_test.go

```diff
@@ -125,12 +126,34 @@ func TestNodeGroupNewNodeGroup(t *testing.T) {
 			return
 		}
 
-		if ng.Name() != tc.name {
-			t.Errorf("expected %q, got %q", tc.name, ng.Name())
+		if ng == nil {
+			t.Fatal("expected nodegroup to be non-nil")
+		}
+
+		var expectedName string
+		var expectedNamespace string
+
+		switch v := (ng.scalableResource).(type) {
+		case *machineSetScalableResource:
+			expectedName = testConfig.spec.machineSetName
+			expectedNamespace = testConfig.spec.namespace
+
+		case *machineDeploymentScalableResource:
+			expectedName = testConfig.spec.machineDeploymentName
+			expectedNamespace = testConfig.spec.namespace
+		default:
+			t.Fatalf("unexpected type: %T", v)
+		}
+
+		expectedID := path.Join(expectedNamespace, expectedName)
```

Comment:
Both MS and MD scalable resource use `path.Join(r.Namespace(), r.Name())` to construct the ID. Would be better to define new private function `constructID` that will wrap the expression. So once the way the ID is generated changes, there is only once place/expression to be changed instead of changing two/three places at once.

Reply:
I only see one case here - it's highly likely that I changed this and force-pushed sometime during the day.